server/src/main/java/io/unitycatalog/server/service/IcebergRestCatalogService.java

```diff
@@ -155,7 +165,7 @@ public HttpResponse tableExists(@Param("namespace") String namespace,
       String metadataLocation =
         tableRepository.getTableUniformMetadataLocation(session, catalog, schema, table);
       if (metadataLocation == null) {
-        return HttpResponse.of(HttpStatus.NOT_FOUND);
+        throw new NoSuchTableException("Table not found: %s", namespace + "." + table);
```

Comment:
minor: we might want to (eventually) update the error msg to `Table does not exist: ...` in case we'd eventually want to run this against the `CatalogTests` from Iceberg (as that's the error that those tests are expecting)